### DISCUSSION TITLE
What did I break...  D1 Mini Restarting every 30 seconds

### DISCUSSION THREAD

**Discussion by jamesonuk** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#806] [General]
I had everything working a while ago but the tub had been mothballed over the summer.
Earlier I downloaded the latest version and uploaded but this is now restarting every 30 seconds.

I have moved the device inside to rule out weak Wi-Fi and RSSI is @ -64 right now but every 30 seconds it just restarts.

Hooking up a serial monitor and I am seeing

```
NTP > synced: 2024-10-13 20:15:41. Saving boot info.

--------------- CUT HERE FOR EXCEPTION DECODER ---------------

Exception (9):
epc1=0x40223024 epc2=0x00000000 epc3=0x401019fb excvaddr=0x0000020a depc=0x00000000

LoadStoreAlignmentCause: Load or store to an unaligned address
  epc1=0x40223024 in Ticker::_static_callback() at ??:?
  epc3=0x401019fb in stopWaveform_weak at ??:?
```
with a stack dump.

Any help with debugging this or pointers on what I have done wrong?

**Comment by visualapproach** [comment]
What version did you download?

**Reply by jamesonuk** [reply]
> What version did you download?

It was a pull of Master from Saturday night.

**Reply by jamesonuk** [reply]
Did make me realise that as I am seeing this without plugging it in, it is quite easy to just try a few versions.
so far 4.4.5, 4.4.2 both exhibited the same.   4.3.0 is working (uptime is now 2:55 and would never get past 0:30 before)

Will do some digging on versions and changes and see if I can work out what it is that is making the D1 unstable

**Edit:** 4.4.1 works fine, 4.4.2 does not.   Will see if I can work out which change is causing my issues

**Reply by visualapproach** [reply]
Ok see if it works on a nodemcu if you have one. Or maybe the board setting must be changed. I read that there is a bug in some definition for 8MB boards so you need to set something in platformio.ini. Can't remember exactly but some file ...eagle_8M_1M... somethingsomething. I think I read it in this forum but I usually don't pay too much attention on the mini d1 stuff ;)

**Comment by jamesonuk** [comment]
OK so I tried on a nodemcu and still the same.   I actually ended up buying a prebuilt device with the LLM but still the same when flashed to this version.

I managed to grab a bit of the dump which I think is probably the issue
```
0x4023901c in esp8266::polledTimeout::timeoutTemplate<false, esp8266::polledTimeout::YieldPolicy::DoNothing, esp8266::polledTimeout::TimePolicy::TimeUnit<esp8266::polledTimeout::TimePolicy::TimeSourceMillis, 1000ull> >::expiredOneShot() const at ??:?
0x40239528 in esp8266::MDNSImplementation::MDNSResponder::_updateProbeStatus() at ??:?
0x4021ca3d in esp8266webserver::ESP8266WebServerTemplate<WiFiServer>::handleClient() at ??:?     
0x4027866a in wifi_get_opmode at ??:?
0x4027984f in wifi_station_get_connect_status at ??:?
0x4021daa0 in ESP8266WiFiSTAClass::status() at ??:?
```

There are various mDNS issues so I suspect this is possibly due to my network config.
The device is running on a VLAN which causes issues with mDNS but not sure what the issue it would cause.


My hunch is around https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/pull/762/commits/5635d2d29b90b6335e01f5611ab2317c7c6e95ce as that seems to be around network connectivity but I haven't managed to get my head around what the change was yet

**Reply by visualapproach** [reply]
Ok, it seems to be an issue with the core libraries which makes it hard to debug. (It's not my code)
Or possibly I use it in an unexpected way or something. Also difficult to track since it works for me.
Maybe the ESP in AP mode is not working in your case. Try hardcoding your SSID and PSWD so the AP never starts. Or comment out those functions.

**Reply by jamesonuk** [reply]
Given I can reproduce on a board with nothing else attached it is easy enough to do some testing.
It seems as if it is doing something odd when getting WiFi details and is them deciding it needs to start the local AP and then decides it can connect and then gets stuck in a loop.

Stuck on `Firmware: 2024-07-07-2211` until I sort it.   Just need some free time to actually look....

**Reply by jamesonuk** [reply]
I have confirmed it is e37740121ce9bd64a86d5a3177271449572fb793 that breaks it.
May get some time over the weekend to look in more depth.   This does kind of tie in with what I am seeing (purely going off commit description).

Says it will restore state after 30s and I am seeing it die every 30s.

**Reply by visualapproach** [reply]
Okay good to know. Would be nice if you track down what it is and maybe we can work around it.

**Comment by jimbo425** [comment]
How to configure d1 mini to Firmware: 2025-07-28-1200 ?
Phisically:
cio d4,d3,d2
dsp d7,d6,d5,d0

How to configure pins in the web page?

**Reply by visualapproach** [reply]
Off topic, but it is described in the manual
